### ISSUE TITLE
repair_fanout7 regression is left with 2 floating iterms after repair_timing

### ISSUE THREAD

**Issue by openroadie** [The-OpenROAD-Project/OpenROAD#3924]
### Describe the bug

Unrelated to gate cloning (did not find any other places where nets with connections were deleted so far). Code to see floating terms:

```
foreach iterm [[ord::get_db_block] getITerms] {
    if { [ $iterm getIoType ] == "INPUT" && [ $iterm getNet ] == "NULL" } {
	puts "Floating iterm: [[$iterm getInst] getName]/[[$iterm getMTerm] getName]"
    }
}
```

### Expected Behavior

No floating terms. 

### Environment

```shell
Not needed
```

### To Reproduce

src/rsz/test/repair_fanout7.tcl with code to dump floating iterms at the end of the script.

### Relevant log output

```shell
N/A
```

**Comment by openroadie** [comment]
cc: @gadfort @maliberty

**Comment by rovinski** [comment]
Appears to be resolved by #3926
